### ISSUE TITLE
False positive [override] check, superclass signature wrong

### ISSUE THREAD

**Issue by pebenito** [python/mypy#16437]
**Bug Report**

mypy is using an incorrect method signature on [override] checks. The code looks like this:

```python
from PyQt6 import QtCore, QtGui, QtWidgets

class SEToolsTreeWidget(QtWidgets.QTreeWidget):

    """QTreeWidget class extended for SETools use."""

    def contextMenuEvent(self, event: QtGui.QContextMenuEvent) -> None:
        ...
```
Full code: https://github.com/pebenito/setools/blob/pyqt-rework/setoolsgui/widgets/views/treewidget.py

When inspecting the PyQt6 definitions, the parameter is not optional:
```
$ grep 'def contextMenuEvent' /usr/lib64/python3.11/site-packages/PyQt6/*.pyi
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, a0: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, a0: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, e: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, a0: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, e: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, event: 'QGraphicsSceneContextMenuEvent') -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, event: 'QGraphicsSceneContextMenuEvent') -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, event: 'QGraphicsSceneContextMenuEvent') -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, event: 'QGraphicsSceneContextMenuEvent') -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, event: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, ev: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, a0: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, event: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, contextMenuEvent: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, e: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, a0: QtGui.QContextMenuEvent) -> None: ...
/usr/lib64/python3.11/site-packages/PyQt6/QtWidgets.pyi:    def contextMenuEvent(self, e: QtGui.QContextMenuEvent) -> None: ...
```

**Expected Behavior**

No errors

**Actual Behavior**

```
setoolsgui/widgets/views/treewidget.py:15: error: Argument 1 of "contextMenuEvent" is incompatible with supertype "QAbstractScrollArea"; supertype defines the argument type as "QContextMenuEvent | None"  [override]
        def contextMenuEvent(self, event: QtGui.QContextMenuEvent) -> None:
                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
setoolsgui/widgets/views/treewidget.py:15: note: This violates the Liskov substitution principle
setoolsgui/widgets/views/treewidget.py:15: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
setoolsgui/widgets/views/treewidget.py:15: error: Argument 1 of "contextMenuEvent" is incompatible with supertype "QWidget"; supertype defines the argument type as "QContextMenuEvent | None"  [override]
        def contextMenuEvent(self, event: QtGui.QContextMenuEvent) -> None:
                                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Example run: https://github.com/pebenito/setools/actions/runs/6804093923/job/18500854778#step:8:36

**Your Environment**

This is running in tox. I hit this both on my Fedora 38 system and also GitHub actions (Ubuntu). tox.ini if useful: https://github.com/pebenito/setools/blob/pyqt-rework/tox.ini

- Mypy version used: 1.6.0, 1.6.1.
- Mypy command-line flags: only `-p` to specify package
- Mypy configuration options from `mypy.ini` (and other config files): from pyproject.toml
```
[tool.mypy]
no_implicit_optional = true
pretty = true

[[tool.mypy.overrides]]
module = ['networkx.*',
          'pytestqt.*']
ignore_missing_imports = true

[[tool.mypy.overrides]]
module = ['tests-gui.*']
disable_error_code = ["union-attr"]
```

- Python version used: 3.10-3.12

**Comment by pebenito** [comment]
This is pure speculation, but the root cause of this might be the same as #16429.

**Comment by pebenito** [comment]
Seen in 1.7.0: https://github.com/pebenito/setools/actions/runs/6851507057/job/18627922726#step:8:37

**Comment by asarkar** [comment]
Given close to 2500 open issues, chances are no one will look at this ticket before you retire.

**Comment by JelleZijlstra** [comment]
I looked at this a little but couldn't find anything. I suspect mypy may be looking at a different set of types for PyQt6 than you're using.

I don't know what #16429 has to do with this issue.
